tests/storage/test_storage.py

```diff
@@ -436,6 +437,18 @@ def test_fetch_file_contents_using_s3_select_and_expect_output_in_csv_format(sel
             OutputSerialization={"CSV": {"FieldDelimiter": ",", "RecordDelimiter": "\n"}},
         )
 
+    @mock.patch.object(storage.S3FileStore, "_get_boto_object_for_key")
+    def test_get_last_modified(self, get_boto_object_for_key):
+        k = mock.Mock()
+        k.last_modified = datetime.datetime.now()
+        get_boto_object_for_key.return_value = k
+        store = storage.S3FileStore("bucket")
+
+        last_modified = store.get_last_modified("a/b/c.pdf")
+        assert get_boto_object_for_key.called
+        assert last_modified == k.last_modified
+        assert type(last_modified) == datetime.datetime
```

Comment:
❓ if we're verifying it's equal above why do we need to check the type? That feels like a type check situation and what's an example where a datetime (which we set in the mock) compares `==` with something that is _not_ a datetime?

In Python 2 that _could_ work in weird scenarios, but in Python 3 comparing types that don't match results in `TypeError` if memory serves.

Reply:
Because I want to make sure nothing is funky happens inside the get_last_modified function that screws up with the return type.  Plus I wanted to be explicit in the test that the type I'd like return is a datetime